app/controllers/adjustments_controller.rb

```diff
@@ -55,7 +55,7 @@ def create
       load_form_collections
       render :new
     end
-  rescue Errors::InsufficientAllotment => e
+  rescue Errors::InsufficientAllotment => e   # TODO:  Can we get rid of this, if we are checking for sufficient allotment in the adjustment validation?
```

Comment:
Hmm... there's a difference between the definition of an adjustment and the application of that adjustment. The errors would be coming from the service rather than the validation. In fact, I'm not sure we even _should_ be validating this kind of thing on the model side - it should be the service methods that do it.

Reply:
How do you distinguish what should be done on the model versus what should be done in the service methods?  (That's an honest question -- what are the criteria?)

Reply:
In general a model should be restricted to the fields that get saved or loaded from the database. Ideally there should be relatively little logic in the model class - it should do some input munging perhaps (formatting strings) etc. It's probably OK to make it a bit fatter in adding some query methods to it, although some want those split out as well. A good rule of thumb seems to be that once we're dealing with more than one model (in this case, the adjustment itself and its line models), the logic shouldn't be in either of the model classes but a service class that works with both of them.

Reply:
But in MVC, "business logic goes in the model",  no?  and "you can't adjust the inventory below 0"  comes under the heading of business logic.     Or am I working from an outdated *ahem* model of how it all works?   Or do we basically consider services to be part of the model part of MVC  (they *feel* more like part of the controller)

Reply:
(nods)  Ok..   I can work with a mental model where the services are part of the model as far as MVC is concerned.   I'll take another look at this with that in mind.